File: data-ingestion/src/db/models/horse.py
"""Horse model."""
from sqlalchemy import Column, String, Integer, Table, ForeignKey
from sqlalchemy.orm import relationship
from src.db.base import Base

# Association table for horse-owner many-to-many relationship
horse_owners = Table(
    'horse_owners',
    Base.metadata,
    Column('horse_id', Integer, ForeignKey('racing.horses.id'), primary_key=True),
    Column('owner_id', Integer, ForeignKey('racing.owners.id'), primary_key=True),
    schema='racing'
)


class Horse(Base):
    """Horse."""

    __tablename__ = "horses"
    __table_args__ = {'schema': 'racing'}

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100), nullable=False, index=True)
    registration_number = Column(String(50), unique=True, index=True)

    # Breeding
    sire_name = Column(String(100))
    dam_name = Column(String(100))
    dam_sire_name = Column(String(100))
    breeder_name = Column(String(200))

    # Breed
    breed = Column(String(50))  # Thoroughbred, Quarter Horse, etc.

    # Relationships
    runners = relationship("Runner", back_populates="horse")
    owners = relationship("Owner", secondary=horse_owners, back_populates="horses")

    def __repr__(self):
        return f"<Horse(name='{self.name}')>"

File: backend-api/src/main/java/com/horseracing/api/entity/Horse.java
package com.horseracing.api.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "horses", schema = "racing")
@Data
@NoArgsConstructor
public class Horse {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "registration_number")
    private String registrationNumber;

    @Column(name = "sire_name")
    private String sireName;

    @Column(name = "dam_name")
    private String damName;

    @Column(name = "dam_sire_name")
    private String damSireName;

    @Column(name = "breeder_name")
    private String breederName;

    @Column(name = "breed")
    private String breed;
}